rosbag2_py/src/rosbag2_py/_info.cpp

```diff
@@ -42,15 +50,29 @@ class Info
     return info_->read_metadata(uri, storage_id);
   }
 
-  void print_output(const rosbag2_storage::BagMetadata & metadata_info)
+  void print_output(
+    const rosbag2_storage::BagMetadata & metadata_info, std::string sorting_method)
   {
+    rosbag2_py::SortingMethod sort_method = rosbag2_py::stringToSortingMethod[sorting_method];
     // Output formatted metadata
-    std::cout << format_bag_meta_data(metadata_info) << std::endl;
+    std::cout << format_bag_meta_data(metadata_info, {}, false, false, sort_method) << std::endl;
   }
 
   void print_output_topic_name_only(const rosbag2_storage::BagMetadata & metadata_info)
   {
-    for (const auto & topic_info : metadata_info.topics_with_message_count) {
+    std::vector<size_t> sorted_idx(metadata_info.topics_with_message_count.size());
+    std::iota(sorted_idx.begin(), sorted_idx.end(), 0);
+    std::sort(
+      sorted_idx.begin(),
+      sorted_idx.end(),
+      [&metadata_info](size_t i1, size_t i2) {
+        std::string topic_name_1 = metadata_info.topics_with_message_count[i1].topic_metadata.name;
+        std::string topic_name_2 = metadata_info.topics_with_message_count[i2].topic_metadata.name;
+        return topic_name_1 < topic_name_2;
```

Comment:
Using `const auto &` is more efficient and convenient. Or at least `const string &`
```suggestion
      [&metadata_info](size_t i1, size_t i2) {
        const auto & topic_name_1 = metadata_info.topics_with_message_count[i1].topic_metadata.name;
        const auto & topic_name_2 = metadata_info.topics_with_message_count[i2].topic_metadata.name;
        return topic_name_1 < topic_name_2;
```